conda/e3sm_diags_env_osx.yml

```diff
@@ -33,6 +33,8 @@ dependencies:
   - esmf=8.0.1
   - esmpy=8.0.1
   - expat=2.2.9
+  - flake8=3.8.4
```

Comment:
@chengzhuzhang mentioned in a comment on `conda/e3sm_diags_env_dev.yml` that we probably don't need to add the dependencies to the user (i.e., non-developer) environment `yml` file.

Reply:
Thanks for the catch, this was accidental.